### ISSUE TITLE
Linux delete and backspace keys copy characters to Primary clipboard

### ISSUE THREAD

**Issue by jdoss** [zed-industries/zed#14362]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When running the nightly or normal build on Fedora 40 with X11 and when I use delete or backspace to remove characters Zed copies them to the clipboard. I use https://github.com/oae/gnome-shell-pano as my clipboard manager and I use Gpaste to sync my clipboard to the primary selection.

### Environment

```
Zed: v0.142.0 (Zed Nightly)
OS: Linux X11 fedora 40
Memory: 251.5 GiB
Architecture: x86_64
```

and

```
Zed: v0.143.6 (Zed)
OS: Linux X11 fedora 40
Memory: 251.5 GiB
Architecture: x86_64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Here's a screen cap of the bug:

[Screencast_07_12_2024_02:38:16_PM.webm](https://github.com/user-attachments/assets/5b030a0b-851b-4eb2-a319-16c712b85b2d)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by culebron** [comment]
Ubuntu 23.10, Mate 1.26.2, can't reproduce this bug.

**Comment by jdoss** [comment]
@culebron Are you sure you are looking at the Primary clipboard? I just turned off the sync from Gpaste and Pano no longer shows the individual characters since it is not syncing things between the clipboard buffers. 

I believe this is a bug that impacts the Primary clipboard.

[Screencast_07_13_2024_10:58:08_AM.webm](https://github.com/user-attachments/assets/74532a7c-5b34-4b0e-887e-acaed8fe961e)

**Comment by culebron** [comment]
@jdoss oh, I see, this has to do with this particular clipboard. Nevermind, I don't use it.

**Comment by CharlesChen0823** [comment]
I thought this might intend behaviour, [`editor.rs`](https://github.com/zed-industries/zed/blob/59ce3535d38946acaa7f7dcf6fcdf0aedd127432/crates/editor/src/editor.rs#L2264C6-L2271C39).

**Comment by andygeorge** [comment]
Still able to reproduce this in `0.172.10` (`b487d48b65a41ef9c7df4892f8f3212530646dc6`). I use this buffer all the time, and to have a simple backspace trash it is pretty frustrating!

**Comment by notpeter** [comment]
> I use Gpaste to sync my clipboard to the primary selection.

@jdoss To clarify, this behavior means that when you make a selection it's effectively the same as a select + copy? (clipboard automatically selection) How do you enable this mode? (so I can try and reproduce) 

@andygeorge Do you also use this mode? If you disable this mode, does the issue still happen or does it only occur because Zed is updating the selection when it deletes a single character?

**Comment by jdoss** [comment]
> > I use Gpaste to sync my clipboard to the primary selection.
> 
> [@jdoss](https://github.com/jdoss) To clarify, this behavior means that when you make a selection it's effectively the same as a select + copy? (clipboard automatically selection) How do you enable this mode? (so I can try and reproduce)

Correct. If you highlight text it will sync it to the copy (ctrl+c) buffer. A few clipboard managers have a feature that will sync the two buffers so you have the same text in the primary (highlight to copy, middle click to paste) and the ctrl+c / ctrl+v buffer. Here's gpaste's preference panel for syncing the two buffers. If you install gpaste you should be able to sync the two buffers using this setting.

![Image](https://github.com/user-attachments/assets/c8bce1ef-0536-4cd2-8866-6c39ae9f6bf3)

To be clear, this isn't a bug with using the clipboard sync. Even without syncing the clipboards you get single characters on backspace in the primary clipboard buffer on each backspace press.

**Comment by andygeorge** [comment]
I am on Linux Mint, using Cinnamon, and don't use GPaste. The "backspace adds the deleted character to the middle-click buffer" is what I am experiencing.

**Comment by notpeter** [comment]
I believe this has been fixed.  If you're still seeing this please comment with details and I'm happy to reopen. 
Thanks all!
